utils/initialize_functions_aws.py

```diff
@@ -0,0 +1,153 @@
+import boto3
+import botocore
+import re
+import ipaddress
+import logging
+
+
+def validate_s3_buckets(cfg: dict):
+    """Function to validate the passed S3 bucket names in the config file for a project"
+
+    Parameters
+    ----------
+    cfg : dict
+        The passed config to be checked for buckets
+    """
+    logging.info("Validating bucket names provided in project config")
+
+    # -- 1. Get bucket names
+    BUCKETS = ["store_bucket", "site_bucket", "data_bucket"]
+    list_buckets = [cfg[bucket] for bucket in BUCKETS if cfg[bucket] != ""]
+
+    # -- 2. Validate
+    for bucket_name in list_buckets:
+        validate_bucket_name(bucket_name)
+
+
+def validate_bucket_name(bucket_name: str):
+    """This function validates a provided name for an S3 bucket, based on the rules
+    outlined at 
+    https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html.
+
+    Parameters
+    ----------
+    bucket_name : str
+        Name to be used for an S3 bucket
+    """
+    logging.debug(f"Check validity of bucket name {bucket_name}")
+    s3_resource = boto3.resource("s3")
+
+    # -- 1. Bucket name too short
+    assert len(bucket_name) >= 3, (
```

Comment:
I feel these two assertions could be one but this is minor `3 <= len(bucket_name) <=64`

Reply:
Made it into two so you get two separate error messages, depending on whether the name is too long or too short